File: python/844. Backspace String Compare/solution.py
class Solution:
    def backspaceCompare(self, s: str, t: str) -> bool:
        return self.generate_string(s) == self.generate_string(t)

    def generate_string(self, string : str):
        stack = []

        for char in string:
            if char != '#':
                stack.append(char)
            elif stack:
                stack.pop()
        return str(stack)

File: java/844. Backspace String Compare/Solution.java
class Solution {
    public boolean backspaceCompare(String s, String t) {
        return generateString(s).equals(generateString(t));
    }

    private String generateString(String str) {
        Stack<Character> stack = new Stack<>();

        for (char c : str.toCharArray()) {
            if (c != '#') stack.push(c);
            else if (!stack.isEmpty()) stack.pop();
        }
        return String.valueOf(stack);
    }
}